examples/yaml-to-anydata/yaml_to_anydata.bal

```diff
@@ -0,0 +1,51 @@
+import ballerina/data.yaml;
+import ballerina/io;
+
+type ServerConfig record {|
+    string host;
+    int port;
+    string protocol;
+|};
+
+final string yamlString = "host: \"localhost\"\n" +
+        "port: 8080\n" +
+        "protocol: \"http\"\n" +
+        "database:\n";
+
+public function main() returns error? {
+    // Convert the YAML string to a record type.
+    ServerConfig serverConfig1 = check yaml:parseString(yamlString);
+    io:println(serverConfig1);
+
+    byte[] yamlByteArr = yamlString.toBytes();
+    // Convert the YAML byte array to a record type.
+    ServerConfig serverConfig2 = check yaml:parseBytes(yamlByteArr);
+    io:println(serverConfig2);
+
+    stream<byte[], error?> byteBlockStream = new (new ByteBlockGenerator(yamlString));
+    // Convert the YAML byte block stream to a record type.
+    ServerConfig serverConfig3 = check yaml:parseStream(byteBlockStream);
+    io:println(serverConfig3);
+}
+
+// Defines a class called `ByteBlockGenerator`, which implements the `next()` method.
```

Comment:
More helpful to introduce as a stream implementor.

"a class called `ByteBlockGenerator`" is self-explanatory?